src/main/java/org/sonar/squidbridge/annotations/RuleTemplate.java

```diff
@@ -24,6 +24,11 @@
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
+/**
+ * @deprecated since 2.7. Will be removed without alternative.
+ * Rule template should be manually registered as being template rule, when defining repositories.
```

Comment:
We should specify which method in the SQ API should be used. Same thing for other annotations.